specs/chain/hs/src/Cardano/Spec/Chain/STS/Rule/Chain.hs

```diff
@@ -164,17 +143,28 @@ instance HasTrace CHAIN where
     -- those values here. The upper bound is arbitrary though.
     mHSz <- Gen.integral (Range.constant 0 4000000)
     mBSz <- Gen.integral (Range.constant 0 4000000)
+
     mTxSz <- Gen.integral (Range.constant 0 4000000)
     mPSz <- Gen.integral (Range.constant 0 4000000)
-    -- The delegation liveness parameter is arbitrarily determined.
-    d <- SlotCount <$> Gen.integral (Range.linear 0 10)
-    -- The size of the rolling widow is arbitrarily determined.
-    w <- Gen.integral (Range.linear 0 10)
+
+    -- The size of the rolling window should be equal to k.
+    -- TODO: we need to adapt the formal specs to reflect this.
+    w <- Gen.integral (Range.linear 1 10)
+
+    -- TODO: The delegation liveness parameter is equal to `2 * k`.
+    -- We should adapt the specs to reflect this.
+    d <- pure $! SlotCount $ 2 * fromIntegral w
+
     -- The percentage of the slots will typically be between 1/5 and 1/4,
     -- however we want to stretch that range a bit for testing purposes.
-    t <- Gen.double (Range.constant (1/6) (1/3))
-    -- The slots per-epoch is arbitrarily determined.
-    spe <- SlotCount <$> Gen.integral (Range.linear 1 1000)
+    t <- pure (1/5) -- Gen.double (Range.constant (1/6) (1/3))
+    -- TODO: we make this a constant till we solve the problem with the number
+    -- of byzantineNodes being a constant in the implementation.
```

Comment:
I really find it confusing that we make reference to "the number of byzantine nodes" here. `t` is fixed in the implementation, and _also, confusingly_ derived from some other thing we call "the number of byzantineNodes". But maybe I'm just being slow :-)

Reply:
Yeah, we need to revise this, also in the implementation. I'll make an issue for this.